tests/ssl-opt.sh

```diff
@@ -598,15 +598,19 @@ maybe_adapt_for_psk() {
     adapt_cmd_for_psk SRV_CMD "$SRV_CMD"
 }
 
-# PSK_PRESENT="YES" if at least one protocol versions supports pure-PSK.
+# PSK_PRESENT="YES" if at least one protocol versions supports at least
+# one PSK key exchange mode.
 PSK_PRESENT="NO"
-# PSK_ONLY="YES" if no key exchange is enabled except pure-PSK.
+# PSK_ONLY="YES" if all the available key exchange modes are PSK-based
+# (pure-PSK or PSK-ephemeral, possibly both).
 PSK_ONLY=""
 for c in $CONFIGS_ENABLED; do
     case $c in
         MBEDTLS_KEY_EXCHANGE_PSK_ENABLED) PSK_PRESENT="YES";;
+        MBEDTLS_KEY_EXCHANGE_PSK_*_ENABLED) PSK_PRESENT="YES";;
```

Comment:
I think you mean `MBEDTLS_KEY_EXCHANGE_*_PSK_ENABLED` here: in 1.2 it's XXX-PSK not PSK-XXX.

Yes, it's inconsistent with 1.3. The 1.2 names come from the standard names of the ciphersuites. I just checked RFC 8446 and the 1.3 names don't seem to come from the standard, though the standard tends to say "PSK with (EC)DHE" or `psk_dhe_ke` so indeed the other order. Sigh.